structures/freighter.py

```diff
@@ -27,20 +30,29 @@ def pop(self) -> str:
         returns an empty string if the containers stack is empty, instead 
         of an error
         """
-        # TODO implement pop()
-        return ''
+        # we are telling the code that if the list self.containers is empty we want it to return an empty string
+        # if the list isn't empty we want it to return and take out the last object put into the list
+        if self.containers == []:
+            return ''
+        else:
+            return (self.containers.pop(-1))
 
     def top(self) -> str:
         """reads and returns the name of the container on top of the stack"""
-        # TODO implement top()
-        return ''
+        # we are identifing a new variable and making it the last object in the new self.containers list
+        dom = self.containers[-1]
```

Comment:
nice use of slicing, beware of variable names that do not serve as good descriptors 🙄